optaplanner-core/src/main/java/org/optaplanner/core/config/solver/termination/TerminationConfig.java

```diff
@@ -520,21 +520,27 @@ public Long calculateUnimprovedTimeMillisSpentLimit() {
     @XmlTransient
     public boolean isConfigured() {
         return terminationClass != null ||
+                terminationCompositionStyle != null ||
```

Comment:
Does this mean that termination is configured? Having just the composition style and nothing else still means there is no actual termination, right?

Reply:
It was missing scoreCalculationCount before, so I copied and pasted all fields in prop order. I think in most cases, if at least one property is set, then there a high probability that an actual termination property is set. `isConfigured` returning true does not guarantee it will terminate. So either, we pick a subset of properties, increasing maintenance cost (since we need to know if a missing property should be checked or not), or we include everything, reducing maintenance cost but allowing some unconfigured termination to return true.

Reply:
@Christopher-Chianelli That is an interesting trade-off. In my opinion, we should prefer correctness over maintenance costs. In order words: we should only improve maintenance costs if the code is already correct.

Assuming we can centralize this logic in a single place, I vote that we reduce the set of null checks to just the actual terminations. After that, I think this can be merged.